lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -236,6 +239,17 @@ func AddInvoice(ctx context.Context, cfg *AddInvoiceConfig,
 		options = append(options, zpay32.CLTVExpiry(uint64(defaultDelta)))
 	}
 
+	// We start by populating the requested routing hints marking their
+	// channels as used so we won't duplicate them.
+	usedChannels := make(map[uint64]struct{})
+	for i, h := range invoice.RouteHints {
+		if i >= 20 {
+			break
+		}
+		options = append(options, zpay32.RouteHint(h))
+		usedChannels[h[0].ChannelID] = struct{}{}
```

Comment:
I don't think we should index like this - what if `h` is empty? Why is only the first channel added to `usedChannels`?

Reply:
@Crypt-iQ 
Added validation for h is empty, missed that so thanks for catching it.
Regarding indexing, I am thinking only the first hop is our direct channel, am I missing something?